### ISSUE TITLE
[Box] Support variable spacing values for each inset

### ISSUE THREAD

**Issue by ryanoglesby08** [telus/tds-core#429]
Currently, the `Box` component only supports equal values for inset, horizontal inset, or vertical inset. Ex:

```jsx
<Box inset={3}>
  Content
</Box>

<Box horizontal={2} vertical={4}>
  Content
</Box>
```

There may be use cases for different values for top or bottom, or left or right. Such as:

* Set top inset to `spacing-2` and bottom inset to `spacing-2`
* Apply ONLY a bottom inset of `spacing-1`

Not totally sure what the API for this would look like. It may mean adding props for `top`, `right`, `bottom`, and `left`. Or maybe expanding the `inset` prop to accept an object with multiple values.

Looking to work through the necessity of this by seeing more examples and developing an intuitive API.

**Comment by jesdavpet** [comment]
@theetrain Just adding my two cents to keep the ticket from going stale ... I would say this adds value for developers as well as designers, since currently some designs call for this layout behaviour, and it is being accomplished with custom CSS hacks on top of `<Box>` for now.

**Comment by Jeffrey-Chang** [comment]
I would like to follow up on this and also potentially take on the work and PR it back to TDS. Currently, we have certain designs that require us to do certain hacks that require us to do padding on a specific side (top, left, bottom or right). Due to the restrictions given to us by `<Box>`, we are not able to do this because it gives equal padding on the sides that are defined. 

One example that I currently have would be in this design where the padding on top of the link that says `Back to Billing` and near the bottom of the app, near the global footer in this design (https://telus.invisionapp.com/d/main/#/console/12448253/314151501/inspect). You can see that the top is `72px` and the bottom is `64px`. Currently there is no way to do this. Of course we could do 1 of 2 ways to remedy this.

1) Use custom CSS and put the padding ourselves but this would defeat the purpose of a design system
2) We could get the designer to adjust the padding to match each other

This is only 1 example but there could also be other examples within other projects that have come across this

**Comment by theetrain** [comment]
This has been backlogged for the core team to discuss: https://telusdigital.atlassian.net/browse/TDS-987

**Comment by Berjesty** [comment]
Equal padding is a requirement to achieve consistency and same visual treatment for this purpose. If there is an outlier need to allow modifying equal padding in a box, this should be brought up as a use case. In the above reference it's not a use case where a designer needs various padding specs, designs need to be updated to support equal padding top bottom left right.

**Comment by invalidred** [comment]
We had discussion with @Berjesty and we are on board with having independent padding using `<Box />` as there is a need for such level of control in our designs. There is an example mentioned by @Jeffrey-Chang above. We went over github looking for examples where custom padding's were added using `.scss` and to override `<Box />` components equal padding and found this to be an issue faced by nutella, dope jackalope, strikeforce and other teams. Here is a reference 

https://github.com/telus/my-account-usage/search?q=padding&unscoped_q=padding
https://github.com/telus/my-telus-registration/search?q=padding&unscoped_q=padding
https://github.com/telus/my-account-overview/search?q=padding&unscoped_q=padding
https://github.com/telus/my-account-home-phone/search?q=padding&unscoped_q=padding
https://github.com/telus/my-telus-home-internet/search?utf8=%E2%9C%93&q=padding

In an ideal scenario there should be no need to use `.scss` files to override paddings. This is something `<Box />` should support out of box.

**Comment by invalidred** [comment]
Also thinking about how the component props could look like
```
<Box inset={{ top: 1, bottom: 2, right: 2, left: 1 }} />
<Box inset={[1, 2, 2, 1]} />
<Box vertical={{ top: 1, bottom: 2}} />
<Box vertical={[1, 2]} />
<Box horizontal={{ top: 1, bottom: 2}} />
<Box horizontal={[1, 2]} />
```
Thoughts?

**Comment by invalidred** [comment]
Had a conversation with @theetrain and he said that there should be no need to have variable `paddings` on `<Box />` component and that design will need to be updated to support `<Box />`.

**Comment by theetrain** [comment]
That is correct, @invalidred 

The core-box is meant to provide the same set of rules to help designers and developers use spacing consistently. We're happy to discuss complex use cases with your team and see if an experience can be optimized for usage of core-box as-is. If there are enough cases presented to the design system where more control over spacing is necessary, then enhancements to Box can be made to improve development speed.

**Comment by theetrain** [comment]
Closing this due to the team's renewed understanding in spacing. We'll be following the spacing principles of core-box for the foreseeable future and will not provide controls to the padding of each side of the box.

Closes #744
